File: frontend/src/types/overview.ts
export interface MonthSummary {
  month: number
  month_name: string
  income: number
  expenses: number
  investments: number
  net_savings: number
}

export interface ExpenseByCategory {
  category_name: string
  amount: number
}

export interface ExpenseByType {
  expense_type: string
  amount: number
}

export interface InvestmentByRisk {
  risk_type: string
  amount: number
}

export interface InvestmentByType {
  investment_type_name: string
  amount: number
}

export interface YearlyOverview {
  year: number
  total_income: number
  total_expenses: number
  total_investments: number
  net_savings: number
  months: MonthSummary[]
  expense_by_category: ExpenseByCategory[]
  expense_by_type: ExpenseByType[]
  investment_by_risk: InvestmentByRisk[]
  investment_by_type: InvestmentByType[]
}

File: backend/app/schemas/overview.py
from pydantic import BaseModel

MONTH_NAMES = [
    "January", "February", "March", "April", "May", "June",
    "July", "August", "September", "October", "November", "December",
]


class MonthSummary(BaseModel):
    month: int
    month_name: str
    income: float
    expenses: float
    investments: float
    net_savings: float


class ExpenseByCategory(BaseModel):
    category_name: str
    amount: float


class ExpenseByType(BaseModel):
    expense_type: str
    amount: float


class InvestmentByRisk(BaseModel):
    risk_type: str
    amount: float


class InvestmentByType(BaseModel):
    investment_type_name: str
    amount: float


class YearlyOverview(BaseModel):
    year: int
    total_income: float
    total_expenses: float
    total_investments: float
    net_savings: float
    months: list[MonthSummary]
    expense_by_category: list[ExpenseByCategory]
    expense_by_type: list[ExpenseByType]
    investment_by_risk: list[InvestmentByRisk]
    investment_by_type: list[InvestmentByType]
class MonthlyOverview(BaseModel):
    month: int
    year: int
    total_income: float
    total_expenses: float
    total_investments: float
    net_savings: float
